File: MedLabO UI/medlabo_desktop/lib/models/uposlenik/medicinsko_osoblje_registration_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'medicinsko_osoblje_registration_request.g.dart';

@JsonSerializable()
class MedicinskoOsobljeRegistrationRequest {
  String? ime;
  String? prezime;
  bool? isActive;
  int? spolID;
  int? zvanjeID;
  String? userName;
  String? password;
  String? email;
  String? phoneNumber;

  MedicinskoOsobljeRegistrationRequest(
      {this.ime,
      this.prezime,
      this.isActive,
      this.spolID,
      this.zvanjeID,
      this.userName,
      this.password,
      this.email,
      this.phoneNumber});

  factory MedicinskoOsobljeRegistrationRequest.fromJson(
          Map<String, dynamic> json) =>
      _$MedicinskoOsobljeRegistrationRequestFromJson(json);

  Map<String, dynamic> toJson() =>
      _$MedicinskoOsobljeRegistrationRequestToJson(this);
}

File: MedLabO.Models/Requests/MedicinskoOsobljeRegistrationRequest.cs
using MedLabO.Models.CustomAttributes;
using System.ComponentModel.DataAnnotations;

namespace MedLabO.Models.Requests
{
    public class MedicinskoOsobljeRegistrationRequest
    {
        [Required]
        [MaxLength(50)]
        public string? Ime { get; set; }

        [Required]
        [MaxLength(70)]
        public string? Prezime { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [Required]
        [GenderCheck]
        public int SpolID { get; set; }

        [Required]
        public int ZvanjeID { get; set; }

        [Required]
        [MaxLength(30)]
        public string UserName { get; set; }

        [Required]
        [MaxLength(30)]
        public string Password { get; set; }

        public string Email { get; set; }
        public string PhoneNumber { get; set; }
    }
}
